pkg/cloudinit/controlplane_init_test.go

```diff
@@ -92,6 +95,31 @@ func TestNewInitControlPlane(t *testing.T) {
 	), "Some /capi/scripts files are missing")
 }
 
+func TestUserSuppliedBootstrapConfig(t *testing.T) {
+	g := NewWithT(t)
+
+	config, err := cloudinit.NewInitControlPlane(cloudinit.InitControlPlaneInput{
+		BaseUserData: cloudinit.BaseUserData{
+			KubernetesVersion:  "v1.30.0",
```

Comment:
nit: let's make this 1.31. since we don't support lower versions.

Reply:
I'll create a card to update all the strings to 1.31 instead of just this one.